service-base/src/main/java/org/eclipse/hono/service/tracing/SamplerProducer.java

```diff
@@ -195,18 +195,12 @@ Resource resource() {
      */
     @Singleton
     @Produces
-    @IfBuildProperty(name = "quarkus.opentelemetry.enabled", stringValue = "true")
-    Sampler sampler(final TracerRuntimeConfig tracerRuntimeConfig) {
-        final var samplerConfig = tracerRuntimeConfig.sampler;
-        if (samplerConfig.parentBased) {
-            LOG.warn("""
-                    'quarkus.opentelemetry.tracer.sampler.parent-based' set to 'true' - \
-                    custom Hono Sampler will not be applied to child spans""");
-        }
-        final String samplerName = Optional.ofNullable(otelConfig.getString(PROPERTY_OTEL_TRACES_SAMPLER))
-                .orElse(samplerConfig.samplerName);
+    @IfBuildProperty(name = "quarkus.otel.enabled", stringValue = "true")
+    Sampler sampler(final OTelRuntimeConfig otelRuntimeConfig) {
+        final String samplerName = Optional.of(otelConfig.getString(PROPERTY_OTEL_TRACES_SAMPLER))
+                .get();
```

Comment:
@sophokles73 `otelRuntimeConfig` has a method  `getServiceName` which could be used as a backup for `samplerName`. But I am not too sure what this name is so I left it unused for now.

Reply:
I think, the `samplerName` should be obtained like this:
```
final String samplerName = Optional.ofNullable(otelConfig.getString(PROPERTY_OTEL_TRACES_SAMPLER))
        .orElse(SamplerType.ALWAYS_ON.getValue());
```
using `io.quarkus.opentelemetry.runtime.config.build.SamplerType.ALWAYS_ON` as default.

(The `otelRuntimeConfig.serviceName()` contains values like `hono-service-command-router`, whereas the sampler name is e.g. "always_on".)